vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNavItem.java

```diff
@@ -280,11 +302,93 @@ public String getPath() {
      */
     public void setQueryParameters(QueryParameters queryParameters) {
         this.queryParameters = queryParameters;
-        refresh();
+        // Apply new query parameters to the path
+        setPath(getPath());
     }
 
-    private void refresh() {
-        setPath(getPath());
+    /**
+     * Sets the specified path aliases to this item. The aliases act as
+     * secondary paths when determining the active state of the item.
+     *
+     * @param aliases
+     *            the path aliases to set to this item
+     */
+    public void setPathAliases(String... aliases) {
```

Comment:
This adds path aliases to the existing ones, so the method name and JavaDoc should reflect that.

Reply:
If we have `setPathAliases`, does `removePathAliases` and `clearPathAliases` still make sense? To me, they seem like an addition to `addPathAliases`, which we don't have anymore.

Reply:
I agree on removing `removePathAliases. However, I'm not sure on how to handle `clearPathAliases`. We can still remove it, but then should clarify the way to remove them using `setPathAliases(String...)`. Maybe we can set `setPathAliases(Set<String>)` public and remove `setPathAliases(String...)`. What do you think?

Reply:
Removed `removePathAliases`, `clearPathAliases`, and `setPathAliases(String...)`. Also made `setPathAliases(Set<String>)` and updated the tests.